core/src/options.rs

```diff
@@ -0,0 +1,18 @@
+/// Configuration for the engine behaviour
+/// The defaults are optimal so please only modify these if you know deliberately why you are modifying them.
+#[derive(Clone, Copy, Debug)]
+pub struct EngineOptions {
+	/// The maximum number of live queries that can be created in a single transaction
+	pub new_live_queries_per_transaction: u32,
```

Comment:
As this is usually consumed as a `usize` shall we better use the type `usize` rather than `u32` so we never need to convert it?  (I guess the compiler optimisation may take care of that, but still)

Reply:
Yes it is usize, but since this is config I think it is best if we establish the size up-front. But it will translate to usize.